daemon/src/db/closed.rs

```diff
@@ -0,0 +1,1365 @@
+            collaborative_settlement_txs
+        JOIN
+            closed_cfds on closed_cfds.id = collaborative_settlement_txs.cfd_id
+        WHERE
+            closed_cfds.uuid = $1
+        "#,
+        id
+    )
+    .fetch_optional(&mut *conn)
+    .await?;
+
+    Ok(row)
+}
+
+async fn load_cet_settlement(
+    conn: &mut PoolConnection<Sqlite>,
+    id: OrderId,
+) -> Result<Option<Settlement>> {
+    let row = sqlx::query_as!(
+        Settlement::Cet,
+        r#"
+        SELECT
+            commit_txs.txid as "commit_txid: model::Txid",
+            cets.txid as "txid: model::Txid",
+            cets.vout as "vout: model::Vout",
+            cets.payout as "payout: model::Payout",
+            cets.price as "price: model::Price"
+        FROM
+            cets
+        JOIN
+            commit_txs on commit_txs.cfd_id = cets.cfd_id
+        JOIN
+            closed_cfds on closed_cfds.id = cets.cfd_id
+        WHERE
+            closed_cfds.uuid = $1
+        "#,
+        id
+    )
+    .fetch_optional(&mut *conn)
+    .await?;
+
+    Ok(row)
+}
+
+async fn load_refund_settlement(
+    conn: &mut PoolConnection<Sqlite>,
+    id: OrderId,
+) -> Result<Option<Settlement>> {
+    let row = sqlx::query_as!(
+        Settlement::Refund,
+        r#"
+        SELECT
+            commit_txs.txid as "commit_txid: model::Txid",
+            refund_txs.txid as "txid: model::Txid",
+            refund_txs.vout as "vout: model::Vout",
+            refund_txs.payout as "payout: model::Payout"
+        FROM
+            refund_txs
+        JOIN
+            commit_txs on commit_txs.cfd_id = refund_txs.cfd_id
+        JOIN
+            closed_cfds on closed_cfds.id = refund_txs.cfd_id
+        WHERE
+            closed_cfds.uuid = $1
+        "#,
+        id
+    )
+    .fetch_optional(&mut *conn)
+    .await?;
+
+    Ok(row)
+}
+
+async fn insert_event_log(
+    conn: &mut Transaction<'_, Sqlite>,
+    id: OrderId,
+    event_log: EventLog,
+) -> Result<()> {
+    for EventLogEntry { name, created_at } in event_log.0.iter() {
+        let query_result = sqlx::query!(
+            r#"
+            INSERT INTO event_log (
+                cfd_id,
+                name,
+                created_at
+            )
+            VALUES
+            (
+                (SELECT id FROM closed_cfds WHERE closed_cfds.uuid = $1),
+                $2, $3
+            )
+            "#,
+            id,
+            name,
+            created_at
+        )
+        .execute(&mut *conn)
+        .await?;
+
+        if query_result.rows_affected() != 1 {
+            anyhow::bail!("failed to insert into event_log");
+        }
+    }
+
+    Ok(())
+}
+
+/// Obtain the time at which the closed CFD was created, according to
+/// the `event_log` table.
+///
+/// Every closed CFD must have gone through contract setup at some
+/// point. Therefore, we base the creation timestamp on the
+/// `EventKind::ContractSetupStarted` variant.
+async fn load_creation_timestamp(
```

Comment:
We could actually use the logic that is used for `failed_cfds` here as well. Can be done in follow-up.

Reply:
Done with https://github.com/itchysats/itchysats/pull/1916/commits/38c75bf4010bccb508edf0a6e0da8ead1c93409e.